src/test/run-make/extern-flag-fun/Makefile

```diff
@@ -3,14 +3,15 @@
 all:
 	$(RUSTC) bar.rs --crate-type=rlib
 	$(RUSTC) bar.rs --crate-type=rlib -C extra-filename=-a
+	$(RUSTC) bar-alt.rs --crate-type=rlib
 	$(RUSTC) foo.rs --extern hello && exit 1 || exit 0
 	$(RUSTC) foo.rs --extern bar=no-exist && exit 1 || exit 0
 	$(RUSTC) foo.rs --extern bar=foo.rs && exit 1 || exit 0
 	$(RUSTC) foo.rs \
 		--extern bar=$(TMPDIR)/libbar.rlib \
-		--extern bar=$(TMPDIR)/libbar-a.rlib \
+		--extern bar=$(TMPDIR)/libbar-alt.rlib \
```

Comment:
Externs are processed as a single group, so we deduplicate on hash irrespective of the filename.